e2e/playwright/testing-selections.spec.ts

```diff
@@ -528,38 +528,164 @@ const sketch002 = startSketchOn(launderExtrudeThroughVar, seg02)
+    const tangentialArcOppositeEdge: Coords2d = [820, 180]
+    const tangentialArcAdjacentEdge: Coords2d = [893, 165]
+
+    const straightSegmentEdge: Coords2d = [819, 369]
+    const straightSegmentOppositeEdge: Coords2d = [635, 394]
+    const straightSegmentAdjacentEdge: Coords2d = [679, 329]
 
     await page.mouse.move(nothing[0], nothing[1])
     await page.mouse.click(nothing[0], nothing[1])
 
     await expect(page.getByTestId('hover-highlight')).not.toBeVisible()
     await page.waitForTimeout(200)
 
-    await page.mouse.move(extrusionTop[0], extrusionTop[1])
-    await expect(page.getByTestId('hover-highlight').first()).toBeVisible()
-    await page.mouse.move(nothing[0], nothing[1])
-    await expect(page.getByTestId('hover-highlight').first()).not.toBeVisible()
+    const checkCodeAtHoverPosition = async (
+      name = '',
+      coord: Coords2d,
+      highlightCode: string,
+      activeLine = highlightCode
+    ) => {
+      await test.step(`test selection for: ${name}`, async () => {
+        const highlightedLocator = page.getByTestId('hover-highlight')
+        const activeLineLocator = page.locator('.cm-activeLine')
+
+        await test.step(`hover should highlight correct code`, async () => {
+          await page.mouse.move(coord[0], coord[1])
+          await expect(highlightedLocator.first()).toBeVisible()
+          await expect
+            .poll(async () => {
+              const textContents = await highlightedLocator.allTextContents()
+              return textContents.join('').replace(/\s+/g, '')
+            })
+            .toBe(highlightCode)
+          await page.mouse.move(nothing[0], nothing[1])
+        })
+        await test.step(`click should put the cursor in the right place`, async () => {
+          await expect(highlightedLocator.first()).not.toBeVisible()
+          await page.mouse.click(coord[0], coord[1])
+          await expect
+            .poll(async () => {
+              const activeLines = await activeLineLocator.allInnerTexts()
+              return activeLines.join('')
+            })
+            .toContain(activeLine)
+          // check pixels near the click location are yellow
+        })
+        await test.step(`check the engine agrees with selections`, async () => {
+          // ultimately the only way we know if the engine agrees with the selection from the FE
+          // perspective is if it highlights the pixels near where we clicked yellow.
+          await expect
+            .poll(async () => {
+              const RGBs = await u.getPixelRGBs({ x: coord[0], y: coord[1] }, 3)
+              for (const rgb of RGBs) {
+                const [r, g, b] = rgb
+                const RGAverage = (r + g) / 2
+                const isRedGreenSameIsh = Math.abs(r - g) < 3
+                const isBlueLessThanRG = RGAverage - b > 45
+                const isYellowy = isRedGreenSameIsh && isBlueLessThanRG
+                if (isYellowy) return true
```

Comment:
This logic is a bit crude, but does the trick.

Reply:
we definitely need a better dialog with the engine at some point. Otherwise it will start failing after we or user will make some color changes for selected lines